packages/react-components/react-utilities/src/compose/types.ts

```diff
@@ -33,16 +33,16 @@ export type UnknownSlotProps = Pick<React.HTMLAttributes<HTMLElement>, 'children
 /**
  * Helper type for {@link Slot}. Adds shorthand types that are assignable to the slot's `children`.
  */
-type WithSlotShorthandValue<Props extends { children?: unknown }> =
+type WithSlotShorthandValue<Props> =
   | Props
-  | Extract<SlotShorthandValue, Props['children']>;
+  | ('children' extends keyof Props ? Extract<SlotShorthandValue, Props['children']> : never);
```

Comment:
Instead of requiring `children` to exist on `Props` signature, we can just check if the property exists and, if so use it.